### PR TITLE
feat!: new persistence api

### LINKED ISSUE #150: fix: drop defaultLanguage
https://github.com/mateonunez/fastify-orama/blob/3b616916f7192b1e740cc67fb6c740d4d297e59c/index.js#L40C7-L40C22

I can't find any `defaultLanguage` prop to provide to orama:

image

I think we should improve the stemming option instead:

https://docs.oramasearch.com/text-analysis/stemming

### LINKED ISSUE #153: feat: improve persistence usability
https://github.com/mateonunez/fastify-orama/blob/3b616916f7192b1e740cc67fb6c740d4d297e59c/index.js#L28-L30

The first run of the application, the file will never exist until the dev calls `save()`

If we keep this persistence utility (totally pro to this) we should improve the DX

### PR DESCRIPTION
**Pull request by Eomm**
closes #153
closes #150

```
-------------------------------|---------|----------|---------|---------|-------------------
File                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------------------|---------|----------|---------|---------|-------------------
All files                      |     100 |      100 |     100 |     100 |                   
 fastify-orama                 |     100 |      100 |     100 |     100 |                   
  index.js                     |     100 |      100 |     100 |     100 |                   
 fastify-orama/lib/persistence |     100 |      100 |     100 |     100 |                   
  in-file.js                   |     100 |      100 |     100 |     100 |                   
  in-memory.js                 |     100 |      100 |     100 |     100 |                   
-------------------------------|---------|----------|---------|---------|-------------------
```

main changes:

- rename the plugin to `fastify-orama`. The `@fastify/` namespace should be used for official ones
- new persistence API - docs soon
- propagate all the option to orama so we are not hiding new cool features

TODO:

- [x] docs
- [x] ts 🤢 <---- not sure I will do it. LOL tsd is passing..

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,91 +2,182 @@
...
-  persistency: {
-    name: './quotes.json',
-    format: 'json'
-  }
+app.register(fastifyOrama, {
+  schema: {
+    quote: "string",
+    author: "string"
+  },
+  persistency: new PersistenceInFile({
```
Comment (mateonunez) [inline README.md]: Change the name to `persistence`.

File: index.js [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
+    save: undefined
+  }
 
   if (persistence) {
-    dbName = options.persistency?.name || './orama.json'
-    dbFormat = options.persistency?.format || 'json'
-    const databaseExists = existsSync(path.resolve(dbName))
+    db = await persistence.restore()
+
+    // todo: could we check if the db.schema is the same as the one provided in the options?
```
Comment (mateonunez) [inline index.js]: Some considerations on this about the performances on large schemas? Or maybe with binaries format. 

You could open an issue to track this. :muscle:

**Review by mateonunez — COMMENTED** [review]
Hey @Eomm, thank you for this contribution. These changes make more flexible the plugin and improve a lot the DX!

I left a couple of comment but the PR seems good! :muscle:

File: README.md (hunk 1 above)
Comment (Eomm) [reply README.md]: Done

File: index.js (hunk 2 above)
Comment (Eomm) [reply index.js]: Done: https://github.com/mateonunez/fastify-orama/issues/159

**Review by mateonunez — APPROVED** [review]
LGTM!

**Pull request merged**